staging/src/k8s.io/client-go/discovery/discovery_client.go

```diff
@@ -60,6 +60,8 @@ type DiscoveryInterface interface {
 }
 
 // CachedDiscoveryInterface is a DiscoveryInterface with cache invalidation and freshness.
+// The implementations will not retry cache misses, unless the user explicitly
+// calls Invalidate after the miss.
```

Comment:
It feels strange to talk about "implementations" in the interface doc. How about something like "Note that If the ServerResourcesForGroupVersion method returns a cache miss error,  the user needs to explicitly call Invalidate to clear the cache, otherwise the same cache miss error will be returned next time."?